frontend/app/src/components/dialogs/BigDialog.vue

```diff
@@ -19,84 +19,94 @@
+      <div class="big-dialog__content" :style="contentStyle">
         <div class="big-dialog__body">
           <slot v-if="display" />
         </div>
       </div>
-      <v-card-actions class="px-6">
-        <v-progress-linear v-if="loading" indeterminate class="mx-4" />
-        <v-spacer />
-        <v-btn
-          color="primary"
-          depressed
-          outlined
-          text
-          class="big-dialog__buttons__cancel"
-          @click="cancel()"
-        >
-          {{ secondaryAction }}
-        </v-btn>
-        <v-btn
-          data-cy="confirm"
-          :color="themes[confirmType].color"
-          :disabled="actionDisabled"
-          depressed
-          class="big-dialog__buttons__confirm"
-          @click="confirm()"
-        >
-          {{ primaryAction }}
-        </v-btn>
-      </v-card-actions>
+      <v-sheet elevation="5">
+        <v-card-actions>
+          <v-progress-linear v-if="loading" indeterminate class="mx-4" />
+          <v-spacer />
+          <v-btn
+            color="primary"
+            depressed
+            outlined
+            text
+            class="big-dialog__buttons__cancel"
+            @click="cancel()"
+          >
+            {{ secondaryAction }}
+          </v-btn>
+          <v-btn
+            data-cy="confirm"
+            :color="themes[confirmType].color"
+            :disabled="actionDisabled"
+            depressed
+            class="big-dialog__buttons__confirm"
+            @click="confirm()"
+          >
+            {{ primaryAction }}
+          </v-btn>
+        </v-card-actions>
+      </v-sheet>
     </v-card>
   </v-bottom-sheet>
 </template>
 
-<script lang="ts">
-import { defineComponent, PropType } from 'vue';
+<script setup lang="ts">
+import { PropType } from 'vue';
 import { DIALOG_TYPES, themes, TYPE_INFO } from '@/components/dialogs/consts';
+import i18n from '@/i18n';
 
-export default defineComponent({
-  name: 'BigDialog',
-  props: {
-    title: { required: true, type: String },
-    subtitle: { required: false, type: String, default: '' },
-    display: { required: true, type: Boolean },
-    loading: { required: false, type: Boolean, default: false },
-    actionDisabled: { required: false, type: Boolean, default: false },
-    primaryAction: { required: false, type: String, default: 'Confirm' },
-    secondaryAction: { required: false, type: String, default: 'Cancel' },
-    confirmType: {
-      required: false,
-      type: String as PropType<typeof DIALOG_TYPES[number]>,
-      default: TYPE_INFO
-    }
+const props = defineProps({
+  title: { required: true, type: String },
+  subtitle: { required: false, type: String, default: '' },
+  display: { required: true, type: Boolean },
+  loading: { required: false, type: Boolean, default: false },
+  actionDisabled: { required: false, type: Boolean, default: false },
+  primaryAction: {
+    required: false,
+    type: String,
+    default: i18n.tc('common.actions.confirm')
```

Comment:
We can't use `useI18n` since we can't use local variable in defineProps.
The other way is to keep the default null, and create another computed variable to set the default.

Reply:
Go with null and computed then :smile: